squire_sdk/src/server/gathering/hall.rs

```diff
@@ -97,27 +99,31 @@ where
                     let sender = self.gatherings.get_mut(&id).unwrap();
                     let (send, recv) = oneshot_channel();
                     let msg = GatheringMessage::GetTournament(send);
-                    sender.send(msg);
+                    if !sender.send(msg) {
+                        panic!("GatheringMessage::GetTournament failed")
+                    }
                     let tourn = recv.await.unwrap();
                     let _ = persist_reqs.insert(id, tourn);
                 }
 
                 persist_reqs
                     .drain()
-                    .for_each(|(_, tourn)| self.persister.send(tourn));
+                    .for_each(|(_, tourn)| {
+                        if !self.persister.send(tourn) {
+                            panic!("Failed to persist tournament")
+                        };
+                    });
```

Comment:
In this case, maybe we should panic, right? Because the persister panicked.

Reply:
This panic is a good idea, at least for now. Failing to persist tournaments is an unrecoverable error.